velox/dwio/common/FileSink.cpp

```diff
@@ -90,9 +96,7 @@ std::unique_ptr<FileSink> FileSink::create(
       return result;
     }
   }
-  // TODO: remove this fallback once file data sink all switch to use velox
-  // filesystem for io operation.
-  return std::make_unique<LocalFileSink>(filePath, options);
+  VELOX_FAIL("FileSink is not registered for {}", filePath);
```

Comment:
Nice fix! I am also thinking of doing this when last time we run into the similar issue in Meta internally.